File: examples/python-unittest/mycounter.py
class MyCounter:
    def __init__(self):
        self.value = 0

    def inc(self):
        self.value += 1

    def reset(self):
        # self.value = self.value / self.value - 1
        self.value = 0
    
    def count(self):
        return self.value

File: examples/javascript-unittest/mycounter.js
function mycounter(initial_value)
{
    this.value = initial_value

    this.inc = function () {
        this.value += 1
    }

    this.reset = function () {
        // Reset to zero in a not-that-funny way
        this.value = (this.value / this.value) - 1
    }

    this.count = function () {
        return this.value
    }
}
